### DISCUSSION TITLE
wazuh agent ERROR: Unable to connect

### DISCUSSION THREAD

**Discussion by 011248163264** [Security-Onion-Solutions/securityonion#6358] [Unsupported Versions] [answered]
latest security onion iso, installed via usb.
default install
i have added syslog clients and this works perfect, now the next is add my servers via wazuh.
1. I have installed the wazhu agent on the servers and added the wazuh server ip in the config
2. so-allow on the sos server and enabled the option (w) and (r) for the servers, i added the server ips.
3. i try to /var/ossec/bin/agent-auth -m 192.168.1.95 (sos server ip)

2021/11/27 01:02:31 agent-auth: INFO: Started (pid: 16430).
2021/11/27 01:02:31 agent-auth: INFO: Starting enrollment process to server: 192.168.1.95
2021/11/27 01:02:31 agent-auth: ERROR: Unable to connect to 192.168.1.95:1515

**Comment by tomjones1977** [comment]
Can you connect to that port?  Try something like `nc -v 192.168.1.95 1515` and see if it connects, just to eliminate network/firewall issues.

**Comment by 011248163264** [comment]
yes its not connecting, on the agent server the firewall is disabled, and on the so manager i have added the ports via so-allow.

**Reply by tomjones1977** [reply]
If it's not connecting it may not be the SO firewall that's the problem - are there any firewall settings on the server you're trying to connect from?  Try running `tcpdump -n -i XXX port 1515` (where XXX is your management NIC) on SO and either run the registration or netcat again on the client.  See if the traffic is reaching SO.

**Reply by 011248163264** [reply]
$ sudo tcpdump -v -n -i eno2 port 1515
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eno2, link-type EN10MB (Ethernet), capture size 262144 bytes
14:59:16.634096 IP 192.168.1.152.45568 > 192.168.1.95.ifor-protocol: Flags [S], seq 3510587155, win 29200, options [mss 1460,sackOK,TS val 3013685639 ecr 0,nop,wscale 7], length 0
14:59:16.634262 IP 192.168.1.95.ifor-protocol > 192.168.1.152.45568: Flags [R.], seq 0, ack 3510587156, win 0, length 0
14:59:24.257742 IP 192.168.1.152.45574 > 192.168.1.95.ifor-protocol: Flags [S], seq 2098328988, win 29200, options [mss 1460,sackOK,TS val 3013693262 ecr 0,nop,wscale 7], length 0
14:59:24.257886 IP 192.168.1.95.ifor-protocol > 192.168.1.152.45574: Flags [R.], seq 0, ack 2098328989, win 0, length 0

**Reply by 011248163264** [reply]
it is connecting

**Reply by tomjones1977** [reply]
The [R.] is a RST packet from your SO to your client.  The port is showing as closed.  It does look like either a firewall issue on SO or the Wazuh agent isn't running.

What is the result of `so-firewall includedhosts wazuh_authd` and `docker ps | grep wazuh`?

**Reply by 011248163264** [reply]
sudo so-firewall includedhosts wazuh_authd
192.168.1.0/24
192.168.1.152
192.168.1.153

**Reply by 011248163264** [reply]
$ sudo docker ps | grep wazuh
591d2950f09e   lupe:5000/security-onion-solutions/so-wazuh:2.3.90                "/entrypoint.sh"         15 hours ago   Up 15 hours             0.0.0.0:1514-1515->1514-1515/tcp, 0.0.0.0:55000->55000/tcp, 0.0.0.0:1514->1514/udp                                                 so-wazuh

**Reply by tomjones1977** [reply]
I'm afraid I'm out of ideas on this one.  The firewall rules look fine (the /24 would have covered it but the specific IP should work too).  The docker container is listening on the right ports, and the tcpdump shows that the packets are reaching SO, but it's still blocking the traffic.  The only other thing I can suggest is a reboot, and a removal and re-adding of the firewall rules if that doesn't work.

**Reply by 011248163264** [reply]
how can i remove the ips ?

**Reply by 011248163264** [reply]
when i try:
sudo vi /opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml // it shows in the config
     51     wazuh_agent:
     52       ips:
     53         delete: []
     54         insert: [192.168.1.0/24, 192.168.1.152, 192.168.1.153]
     55     wazuh_api:
     56       ips:
     57         delete: []
     58         insert: [192.168.1.0/24]
     59     wazuh_authd:
     60       ips:
     61         delete: []
     62         insert: [192.168.1.0/24, 192.168.1.152, 192.168.1.153]:

**Reply by tomjones1977** [reply]
`so-firewall excludehost wazuh_authd 1.2.3.4` followed by `so-firewall removehost wazuh_authd 1.2.3.4` should remove the IP, then you can re-add it.

**Reply by 011248163264** [reply]
i have removed them and added them again but still not working
     51     wazuh_agent:
     52       ips:
     53         delete: []
     54         insert: [192.168.1.152]
     55     wazuh_api:
     56       ips:
     57         delete: []
     58         insert: [192.168.1.152]
     59     wazuh_authd:
     60       ips:
     61         delete: []
     62         insert: [192.168.1.152]

**Reply by tomjones1977** [reply]
Have you tried rebooting the system?  I can't think what else it could be that's stopping the traffic from getting through.

**Reply by 011248163264** [reply]
yes i have restartet its still the same

/var/ossec/bin/agent-auth -m 192.168.1.95
2021/11/27 17:10:15 agent-auth: INFO: Started (pid: 6720).
2021/11/27 17:10:15 agent-auth: INFO: Starting enrollment process to server: 192.168.1.95
2021/11/27 17:10:15 agent-auth: ERROR: Unable to connect to 192.168.1.95:1515

**Reply by 011248163264** [reply]
i found this one:
sudo service firewalld status
Redirecting to /bin/systemctl status firewalld.service
● firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; enabled; vendor preset: enabled)
   Active: active (running) since Sa 2021-11-27 16:04:22 UTC; 28min ago
     Docs: man:firewalld(1)
 Main PID: 816 (firewalld)
    Tasks: 2
   Memory: 13.6M
   CGroup: /system.slice/firewalld.service
           └─816 /usr/bin/python2 -Es /usr/sbin/firewalld --nofork --nopid

Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -F DOCKER' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -X DOCKER' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -F DOCKER-ISOLATION-STAGE-1' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -X DOCKER-ISOLATION-STAGE-1' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -F DOCKER-ISOLATION-STAGE-2' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -X DOCKER-ISOLATION-STAGE-2' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -F DOCKER-ISOLATION' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:05 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -t filter -X DOCKER-ISOLATION' failed: iptables: No chain/target/match by that name.
Nov 27 16:05:06 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -D FORWARD -i docker0 -o docker0 -j DROP' failed: iptables: Bad rule (does a matching rule exist in that chain?).
Nov 27 16:05:30 lupe firewalld[816]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -D FORWARD -i docker0 -o docker0 -j DROP' failed: iptables: Bad rule (does a matching rule exist in that chain?).
Hint: Some lines were ellipsized, use -l to show in full.

**Reply by 011248163264** [reply]
after restart the firewalld:
sudo service firewalld status
Redirecting to /bin/systemctl status firewalld.service
● firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; enabled; vendor preset: enabled)
   Active: active (running) since Sa 2021-11-27 16:35:14 UTC; 24s ago
     Docs: man:firewalld(1)
 Main PID: 4474 (firewalld)
    Tasks: 2
   Memory: 26.8M
   CGroup: /system.slice/firewalld.service
           └─4474 /usr/bin/python2 -Es /usr/sbin/firewalld --nofork --nopid

Nov 27 16:35:12 lupe systemd[1]: Starting firewalld - dynamic firewall daemon...
Nov 27 16:35:14 lupe systemd[1]: Started firewalld - dynamic firewall daemon.
Nov 27 16:35:15 lupe firewalld[4474]: WARNING: AllowZoneDrifting is enabled. This is considered an insecure configuration option. It will be removed in a future release. Please conside...abling it now.
Nov 27 16:35:17 lupe firewalld[4474]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -D FORWARD -i docker0 -o docker0 -j DROP' failed: iptables: Bad rule (does a matching rule exist in that chain?).
Nov 27 16:35:17 lupe firewalld[4474]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -D FORWARD -i docker0 -o docker0 -j DROP' failed: iptables: Bad rule (does a matching rule exist in that chain?).
Hint: Some lines were ellipsized, use -l to show in full.

**Reply by tomjones1977** [reply]
You could try the following, though it's not guaranteed to work:

```
systemctl stop firewalld
systemctl restart docker
systemctl start firewalld
```

**Reply by 011248163264** [reply]
no changes, its not working

**Reply by tomjones1977** [reply]
Your error seems to be the same as #6352 

I can't think of anything else to check at this stage.  Hopefully one of the devs will have more of an idea why this is happening on the latest version.

**Comment by 011248163264** [comment]
thats my firewall config:
     51     wazuh_agent:
     52       ips:
     53         delete: []
     54         insert: [192.168.1.152, 192.168.1.0/24]
     55     wazuh_api:
     56       ips:
     57         delete: []
     58         insert: [192.168.1.152, 192.168.1.0/24]
     59     wazuh_authd:
     60       ips:
     61         delete: []
     62         insert: [192.168.1.152, 192.168.1.0/24]

there is no firewall from 192.168.1.152 to 192.168.1.95 only the so-firewall

**Comment by dougburks** [comment, marked as answer]
Which ISO image did you download?

**Reply by 011248163264** [reply]
this one https://github.com/Security-Onion-Solutions/securityonion/blob/master/VERIFY_ISO.md

**Reply by 011248163264** [reply]
https://download.securityonion.net/file/securityonion/securityonion-2.3.90-WAZUH.iso

**Reply by dougburks** [reply]
I just tested a fresh installation of securityonion-2.3.90-WAZUH.iso and it appears to be working for me:
![image](https://user-images.githubusercontent.com/1659467/144055627-555f4e9e-813f-44e3-8f14-eef017fc34b1.png)

Please double check that you are using securityonion-2.3.90-WAZUH.iso and not the original securityonion-2.3.90.iso as it had a bug in the Wazuh configuration. Then try a fresh installation using securityonion-2.3.90-WAZUH.iso and see if it works for you.

**Reply by 011248163264** [reply]
ok yes thanks i found the iso i have downloadet it was the securityonion-2.3.90.iso
i will try with the wazuh iso.

**Reply by 011248163264** [reply]
ok its working with the wazuh iso.
